neqo-crypto/src/secrets.rs

```diff
@@ -43,23 +44,16 @@ impl From<SSLSecretDirection::Type> for SecretDirection {
 #[allow(clippy::module_name_repetitions)]
 pub struct DirectionalSecrets {
     // We only need to maintain 3 secrets for the epochs used during the handshake.
-    secrets: [Option<SymKey>; 3],
+    secrets: EnumMap<Epoch, Option<SymKey>>,
 }
 
 impl DirectionalSecrets {
     fn put(&mut self, epoch: Epoch, key: SymKey) {
-        assert!(epoch > 0);
-        let i = (epoch - 1) as usize;
-        assert!(i < self.secrets.len());
-        // assert!(self.secrets[i].is_none());
-        self.secrets[i] = Some(key);
+        self.secrets[epoch] = Some(key);
     }
```

Comment:
Previously this function would panic on `epoch` `Initial`. Is it safe to change that behavior?

Reply:
Ping @martinthomson

Reply:
We can't get secrets for the Initial phase.  While we initialize the packet encryption with something, that is done at connection initialization time and we don't use the secrets store for that.

As for whether it makes sense to try to avoid the extra one value in the map, I don't think so.  The `SymKey` this holds is a simple pointer.